### ISSUE TITLE
Apptainer 1.1.0 doesn't work with lhcathome atlas unlike apptainer 1.0.3

### ISSUE THREAD

**Issue by kotenok2000** [apptainer/apptainer#611]
### Version of Apptainer

What version of Apptainer (or Singularity) are you using? Run

apptainer-1.1.0~rc.1-1
### Expected behavior

Lhc@home atlas native works normally just like on apptainer version 1.0.3-1

[Новый файл1.txt](https://github.com/apptainer/apptainer/files/9308373/1.txt)

### Actual behavior
[Новый файл.txt](https://github.com/apptainer/apptainer/files/9308372/default.txt)

### Steps to reproduce this behavior

Install apptainer-1.1.0~rc.1-1.el8.x86_64.rpm

### What OS/distro are you running

```NAME="Rocky Linux"
VERSION="8.6 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.6 (Green Obsidian)"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky Linux"
ROCKY_SUPPORT_PRODUCT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8"


```

### How did you install Apptainer
From EPEL Testing x86_64
https://centos.pkgs.org/8/epel-testing-x86_64/apptainer-1.1.0~rc.1-1.el8.x86_64.rpm.html

**Comment by DrDaveD** [comment]
Thank you for checking!  However I'm going to need more information than that to reproduce it.  Can you come up with a scenario that can be done outside of lhcathome, or at least show me how to run the lhcathome scripts on my own host?  I do have access to /cvmfs/atlas.cern.ch and tried running a command similar to the one shown to fail on an centos7 host and it did not fail.

**Comment by kotenok2000** [comment]
I have installed it with this guide
https://lhcathome.cern.ch/lhcathome/forum_thread.php?id=4971
My /etc/cvmfs/default.local
[default.local.txt](https://github.com/apptainer/apptainer/files/9308613/default.local.txt)

**Comment by DrDaveD** [comment]
That does not seem to be a complete guide.  The boinc-client.service file is assuming the installation of a /usr/bin/boinc command, for example.  I installed boinc-client from epel but it still failed to start the service.  Is there another guide that's needed as well?

**Comment by kotenok2000** [comment]
try sudo systemctl status boinc-client
 sudo systemctl enable boinc-client
 sudo systemctl start boinc-client

**Comment by kotenok2000** [comment]
I have installed boinc by sudo dnf install boinc*

**Comment by DrDaveD** [comment]
Ok I did that and then boinc-client started.  What then?

**Comment by kotenok2000** [comment]
Run boincmgr

**Comment by kotenok2000** [comment]
Maybe it is because Apptainer no longer installs a setuid-root portion by default?

**Comment by DrDaveD** [comment]
> Maybe it is because Apptainer separated setuid-root portion to apptainer-suid?

It's possible, but then I would still want to know exactly why non-setuid doesn't work.

I run boincmgr, it says "You don't have any projects. Please Add a Project", but it is non-responsive and nothing I click on does anything.  This is on a rocky8 VM running under VirtualBox on a Mac.

**Comment by kotenok2000** [comment]
Can you take a screenshot?

**Comment by DrDaveD** [comment]
![Screen Shot 2022-08-11 at 9 16 17 AM](https://user-images.githubusercontent.com/2129743/184154518-acba9048-4a07-4282-9f32-ae87d3d995c6.png)

**Comment by kotenok2000** [comment]
Can you use boinctui?
Rpc password is at /var/lib/boinc/gui_rpc_auth.cfg

**Comment by DrDaveD** [comment]
Yes, that works.

I added LHC@Home, created a user account for myself, and now it says "VirtualBox is not installed".  I'm not sure it's a good idea to install VirtualBox inside of a VirtualBox VM!  What do you suggest I do next?

**Comment by kotenok2000** [comment]
Looks llike problem was  in not installed apptainer-suid.
11.08.2022 17:41:17, DrDaveD пишет:

**Comment by DrDaveD** [comment]
But I still want to know why apptainer-suid is needed.  It should not be needed.  If I install the native Mac boinc client, will I be able to connect into the VM created and modify it?

**Comment by DrDaveD** [comment]
I found out quite a bit more from David Cameron of ATLAS@Home.  He told me how to connect to the console and log in and also said that you must be using the native mode.  I enabled "run native if available" in my [lhcathome preferences](https://lhcathome.cern.ch/lhcathome/prefs.php?subset=project), increased the RAM & disk space available to my rocky8 VM, and now I'm seeing "Computation error" on the Theory jobs.  I suspect it is requiring a GPU, and I don't have one.

More importantly he told me that he has seen the error message before that you saw about `mkdir /var/lib/condor: Permission denied` and that it has nothing to do with singularity/apptainer.  He said

> I have seen this singularity error myself in the past when running the standard Linux boinc client installation which runs under /var/lib/boinc under the boinc user. I usually work around it by running the client from a clean directory like /home/boinc which doesn't interfere with what is in the container.

So I would really like you to try it again without apptainer-suid.  A rocky8 system ought to have unprivileged user namespaces enabled by default, so the non-suid apptainer really should work.

**Comment by kotenok2000** [comment]
Actually i have installed centos stream 8 and then converted it to rocky 8 using this guide. 
And after i have installed  apptainer-suid errors stopped.
https://docs.rockylinux.org/guides/migrate2rocky/

**Comment by DrDaveD** [comment]
Starting with centos stream 8 shouldn't make any difference. 

I figured out what was causing the 'Computation error' on my machine by looking in /var/lib/boinc/slots/0/stderr.txt.  My guess about a GPU was all wrong.  The problem was that I had enabled cgroups v2 on my machine.  I disabled that in /etc/default/grub, re-made the grub config, and rebooted, and now jobs are running on my VM.  I don't even see it using singularity at all, it is using /cvmfs/grid.cern.ch/vc/containers/runc.  So why is it trying to use apptainer on your machine?

I tried disabling unprivileged user namespaces with `sysctl user.max_user_namespaces=0` and then the jobs immediately exited with `[ERROR] max_user_namespaces in /proc/sys/user/ is set to 0`.  So that must be enabled on your machine and apptainer should work without apptainer-suid.  Are you are somehow running some different types of job based on singularity instead of runc?
